Lib/test/_test_multiprocessing.py

```diff
@@ -3879,6 +3879,130 @@ def test_shared_memory_ShareableList_pickling(self):
         deserialized_sl.shm.close()
         sl.shm.close()
 
+    @unittest.skipUnless(os.name == 'posix', 'posix shared memory')
+    def test_posix_shared_memory_cleaned_after_process_termination(self):
```

Comment:
Giving it a second though, maybe we need only the windows version of this test

Reply:
Why wouldn't need this test work on both Windows and POSIX? Also, why do you need two subprocesses?

Reply:
> Why wouldn't this test work on both Windows and POSIX? 

There are two tests in this PR: one for Windows and one for POSIX. Becauses Windows and POSIX different way of handling files, testing all OS in one test was not very natural.

> Also, why do you need two subprocesses?

The second subprocess acts as a sanity check to verify that the segment is functional at the beginning. Although I agree it is not crucial for this test. I'm ok with removing the second subprocess if you think we should merge this into #13222

Reply:
Ah, maybe I misread your first remark. The POSIX version of this test is somewhat redundant with the tests in #13222, that is why it is non-essential.

Reply:
What I don't understand is: why would the POSIX test not work on Windows? The `SharedMemory` API is supposed to be portable. Instead of passing a fd so that the child writes to it, the child can simply write on stdout.

Reply:
There is not problem with the `SharedMemory` API, the tests separation was only motivated by fds handling.

> Instead of passing a fd so that the child writes to it, the child can simply write on stdout.

Oh. Well.

Reply:
Alghouh I think the interest of using a separate pipe here is to communicate with the child process while it is still alive - We kill the child process later on in the test. Correct me if I am wrong, but trying to read from stdout/stderr will block until process termination right?

Reply:
If you only read the required amount (for example by calling `p.stdout.readline`), it shouldn't. I'll let you give it a try :-)

Reply:
Ah. I was missing a `flush`. Sorry about that.